src/web/src/pages/_app.tsx

```diff
@@ -27,11 +27,27 @@ const App = ({ Component, pageProps }: AppProps) => {
     if (jssStyles) {
       jssStyles.parentElement?.removeChild(jssStyles);
     }
-  }, []);
+
+    // Enable the stylesheet for the syntax highlighting depending on the theme
+    if (preferredTheme === 'dark') {
+      document.querySelector('#light-stylesheet')?.setAttribute('disabled', 'disabled');
+    } else {
+      document.querySelector('#dark-stylesheet')?.setAttribute('disabled', 'disabled');
+    }
+  });
 
   // Switch the active theme, and also store it for next load
+  // We also disable the current stylesheet and enable the other one.
   const toggleTheme = () => {
-    setPreferredTheme(preferredTheme === 'dark' ? 'light' : 'dark');
+    if (preferredTheme === 'dark') {
+      document.querySelector('#light-stylesheet')?.removeAttribute('disabled');
```

Comment:
We're repeating code here, let's consider moving this enable/disable logic to function(s), and maybe even put it in `hooks/use-preferred-theme` so it happens when we get/set the theme.